Packet++/src/SSLHandshake.cpp

```diff
@@ -1228,6 +1228,19 @@ SSLHandshakeMessage* SSLHandshakeMessage::createHandshakeMessage(uint8_t* data,
 		return nullptr;
 
 	ssl_tls_handshake_layer* hsMsgHeader = (ssl_tls_handshake_layer*)data;
+
+	if (dataLen >= 16)
+	{
+		if (*(uint64_t*)data <= 0xFFFFFF || hsMsgHeader->length1 >= 1)
```

Comment:
Are you sure this is correct `*(uint64_t*)data <= 0xFFFFFF`? 🤔 

Don't we want to take the 5 first bytes of the message data (that comes after the message header), e.g `data + sizeof(ssl_tls_handshake_layer)` and check if all of them are equal to 0?

Reply:
How I understood Wireshark's approach, we want the first 5 bytes of the handshake message - including ```ssl_tls_handshake_layer``` in our case - to be zero.
```ssl_tls_handshake_layer``` is namely the additional 4-byte header that comes after the SSL Record Layer Header and specifies the handshake type and payload length. We want to check if these 4 bytes plus the first byte after the header is zero. The expression ```be64toh(*(uint64_t*)data) <= 0xFFFFFF``` should actually do that. I didn't consider the correct byte order, which is now done in the current commit [ea9c0be](https://github.com/seladb/PcapPlusPlus/pull/1087/commits/ea9c0be4dd6cf8b314694612459960d312049cc9) through ```be64toh```.

Reply:
You're right, my bad! 👍